src/Tools/dotnet-monitor/EndpointInfo/ClientEndpointInfoSource.cs

```diff
@@ -26,7 +38,14 @@ public async Task<IEnumerable<IEndpointInfo>> GetEndpointInfoAsync(CancellationT
                 {
                     try
                     {
-                        return await EndpointInfo.FromProcessIdAsync(pid, token);
```

Comment:
Shouldn't each process be given the same timeout? That way if one process hangs, it doesn't abandon reading every subsequent process

Reply:
The timeout _is_ the same for all of the processes. In reality, each process gets _at least_ the timeout's worth of time to finish. The alternative is that I create a CancellationTokenSource for each process and start it for each process, but that would create a timer for each process.

Each of these is run as a separate task so as long as the threadpool is large enough, all of them can run simultaneously.